amplify/backend/function/getKnowledgeGraph/src/index.js

```diff
@@ -39,59 +57,78 @@ exports.handler = async (event) => {
         id: paperInfo.data.paperId,
         label: paperInfo.data.title,
         type: "main",
+        referenceCount,
+        citationCount,
+        authors: getAuthorName(paperInfo.data),
+        published: paperInfo.data.publicationDate,
+        journal: getJournal(paperInfo.data),
+        venue: paperInfo.data.venue,
       },
     });
 
     // Adds a node for each referenced paper and sets up the
     // edges from current paper to all papers it references
     for (i = 1; i < referenceCount + 1; i++) {
       let reference = paperInfo.data.references[i - 1];
-      nodesAndEdges.push({
-        data: {
-          id: reference.paperId,
-          label: reference.title,
-          type: "reference",
-        },
-      });
-      nodesAndEdges.push({
-        data: {
-          source: paperInfo.data.paperId,
-          target: reference.paperId,
-          label: "Edge label placeholder",
-          type: "reference",
-        },
-      });
+      if (reference) {
+        nodesAndEdges.push({
+          data: {
+            id: reference.paperId,
+            label: reference.title,
+            type: "reference",
+            authors: getAuthorName(reference),
+            published: reference.publicationDate,
+            journal: getJournal(reference),
+            venue: reference.venue,
+          },
+        });
+        nodesAndEdges.push({
+          data: {
+            source: paperInfo.data.paperId,
+            target: reference.paperId,
+            label: "Edge label placeholder",
+            type: "reference",
+          },
+        });
+      }
     }
 
     // Adds a node for each citation and sets up the
     // edges from papers which cite this paper to the current paper
     for (i = referenceCount + 1; i < verticesCount; i++) {
       let citation = paperInfo.data.citations[i - referenceCount - 1];
-      nodesAndEdges.push({
-        data: {
-          id: citation.paperId,
-          label: citation.title,
-          type: "citation",
-        },
-      });
-      nodesAndEdges.push({
-        data: {
-          source: paperInfo.data.paperId,
-          target: citation.paperId,
-          label: "Edge label placeholder",
-          type: "citation",
-        },
-      });
+      if (citation) {
+        nodesAndEdges.push({
+          data: {
```

Comment:
As for some nodes appearing with no edge and just containing the text saying "Edge label placeholder", the fix here should be just adding an attribute called `group` to all nodes and edges objects (main node as well). For nodes, this attribute should have the value `nodes` and for edges the value `edges`. This change shouldn't require any additional changes on the frontend, since it just consumes the values received from the backend.